agent/tests/test_rock.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import random
+import pytest
+import string
+import subprocess
+
+from charmed_kubeflow_chisme.rock import CheckRock
+
+
+@pytest.fixture()
+def rock_test_env(tmpdir):
+    """Yields a temporary directory and random docker container name, then cleans them up after."""
+    container_name = "".join(
+        [str(i) for i in random.choices(string.ascii_lowercase, k=8)]
+    )
+    yield tmpdir, container_name
+
+    try:
+        subprocess.run(["docker", "rm", container_name])
+    except Exception:
+        pass
+    # tmpdir fixture we use here should clean up the other files for us
+
+
+@pytest.mark.abort_on_fail
+def test_rock(rock_test_env):
+    """Test rock."""
+    temp_dir, container_name = rock_test_env
+    check_rock = CheckRock("rockcraft.yaml")
+    rock_image = check_rock.get_name()
+    rock_version = check_rock.get_version()
+    LOCAL_ROCK_IMAGE = f"{rock_image}:{rock_version}"
+
+    # assert the rock contains the expected files
+    subprocess.run(
+        [
+            "docker",
+            "run",
+            "--entrypoint",
+            "/bin/bash",
+            LOCAL_ROCK_IMAGE,
+            "-c",
+            "ls -la /third_party",
+        ],
+        check=True,
+    )
+
+    subprocess.run(
+        [
+            "docker",
+            "run",
+            "--entrypoint",
+            "/bin/bash",
+            LOCAL_ROCK_IMAGE,
+            "-c",
+            "ls -la /ko-app/agent",
+        ],
+        check=True,
+    )
```

Comment:
hmm does this work? I tried running those commands manually and noticed that they both return the same results
```bash
$ docker run --entrypoint /bin/bash kserve-agent:0.11.2 -c ls -la /third_party 
bin
boot
dev
etc
home
ko-app
lib
lib32
lib64
libx32
media
mnt
opt
proc
root
run
sbin
srv
sys
third_party
tmp
usr
var

$ docker run --entrypoint /bin/bash kserve-agent:0.11.2 -c ls -la /ko-app/agent
bin
boot
dev
etc
home
ko-app
lib
lib32
lib64
libx32
media
mnt
opt
proc
root
run
sbin
srv
sys
third_party
tmp
usr
var

```

I 'm not a 100% sure this works the same in sanity tests but why wouldn't it. What about modifying this to sth like [this](https://github.com/canonical/seldonio-rocks/blob/c00b52007ea5687c15261442a150a9ef43e9d13d/mlserver-xgboost/tests/test_rock.py#L44)? 
```
docker run kserve-agent:0.11.2 exec ls -la /ko-app/agent
```

Reply:
Also I think we don't need the fixture at all here. I guess we 've included that in other repos where it's not needed (like the example I linked above). It's only needed when we 're actually creating a container like [here](https://github.com/canonical/seldonio-rocks/blob/main/seldon-core-operator/tests/test_rock.py#L50).

Reply:
Hmm actually this last one is an issue with our tests because we don't clean up the created containers at all. I 'll file an issue for this. I 'm OK with merging it here this way but if you feel like fixing it in your PR, that'd be great too.

Reply:
Added a comment about this [here](https://github.com/canonical/bundle-kubeflow/issues/817#issuecomment-2006553828).

Reply:
good catch on the commands ^
I think there are other places where we need to fix this as well

Reply:
> Added a comment about this [here](https://github.com/canonical/bundle-kubeflow/issues/817#issuecomment-2006553828). Thus I think fixing that as part of this PR doesn't really make sense, since we 'd like to rewrite ROCKs test from scratch.

I think it's alright for me to fix it here, since I'm adding new tests and not refactoring existing ones, it makes sense to write the new tests with this in mind.